src/main/java/org/spongepowered/common/entity/EntityUtil.java

```diff
@@ -406,72 +407,69 @@ public static ServerPlayerEntity transferPlayerToWorld(final ServerPlayerEntity
             toWorld = (ServerWorld) event.getToWorld();
         }
 
-        final int dimensionId;
+        final DimensionType dimensionId;
 
         if (!((ServerPlayerEntityBridge) player).bridge$usesCustomClient()) {
 
             // Check if the world we're going to matches our provider type. if so, we need to send a fake respawn packet to clear chunks
-            final int fromClientDimId = WorldManager.getClientDimensionId(player, fromWorld);
-            final int toClientDimId = WorldManager.getClientDimensionId(player, toWorld);
+            DimensionType fromType = fromWorld.getDimension().getType();
+            DimensionType toType = toWorld.getDimension().getType();
 
-            if (fromClientDimId == toClientDimId) {
-                final int fakeDimId;
-                switch (fromClientDimId) {
+            if (fromType == toType) {
```

Comment:
@Zidane is this right with dimension handling? I'd imagine the type could be the same but the dimension id assigned by the WorldManager could be different?

Reply:
This is to send a fake respawn packet to vanilla clients in case the dimension type is not changing.
This entire teleport things needs extensive testing anyways once we can actually start a server.
Maybe its not needed anymore?

Reply:
No you still have to fake sent a dimension change else you must send an unload packet for every chunk.

That is not including the other fun things the client sets based on the dimension.

Reply:
I think this is still mentioned in the protocol wiki